geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloFeatureReader.scala

```diff
@@ -98,6 +100,7 @@ class AccumuloFeatureReaderWithStats(query: Query,
       timings.time("next") + timings.time("hasNext"),
       count
     )
+    Monitoring.log(stat)
```

Comment:
if we put a few more mandatory fields into usage stat, and require it to implement something like `toJson/fromJson`, we might be able to be pretty flexible with the writers being able to handle any subclasses. Anything more complicated like csv, etc we could do as a secondary process as you say.